src/core/numerics/time_integrator/time_integrator_utils.hpp

```diff
@@ -0,0 +1,62 @@
+#ifndef PHARE_CORE_NUMERICS_TIME_INTEGRATOR_UTILS_HPP
+#define PHARE_CORE_NUMERICS_TIME_INTEGRATOR_UTILS_HPP
+
+#include "core/data/grid/gridlayout_utils.hpp"
+#include "core/data/vecfield/vecfield_component.hpp"
+#include "core/utilities/index/index.hpp"
+
+namespace PHARE::core
+{
+template<typename GridLayout>
+class RKUtils : public LayoutHolder<GridLayout>
+{
+    constexpr static auto dimension = GridLayout::dimension;
+    using LayoutHolder<GridLayout>::layout_;
+
+public:
+    template<typename ReturnState, typename... Pairs>
+    void operator()(ReturnState& res, const Pairs... pairs) const
+    {
+        auto result_fields = getFieldTuples_(res);
+
+        auto weight_tuple = std::make_tuple(std::get<0>(pairs)...);
```

Comment:
since they are pairs, you could use `pairs.first` and `pair.second` that would be more readable.
Perhaps even having a struct would be even easier to read because members would have names

```cpp


struct RKPair
{
    double k;
    std::string s;
};



template <typename... RKPairs>
auto rkdo(RKPairs... rkpairs)
{
    auto t = std::make_tuple(rkpairs...);
    
    std::apply([&](auto&... pairs)
    {
        ((std::cout << pairs.k << " " << pairs.s << "\n"),...);   
    }, t);   
}



int main()
{
  rkdo(RKPair{0.25, "step1"}, RKPair{0.5, "step2"}, RKPair{0.25, "step3"});
}


```